lib/Parser/Scan.cpp

```diff
@@ -1723,28 +1728,35 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
                 Assert(chType == _C_DIG || chType == _C_DOT);
                 p = m_pchMinTok;
                 this->RestoreMultiUnits(m_cMinTokMultiUnits);
-                bool likelyInt = true;
-                pchT = FScanNumber(p, &dbl, likelyInt, savedMultiUnits);
+                LikelyNumberType likelyType = LikelyNumberType::Int;
+                pchT = FScanNumber(p, &dbl, likelyType, savedMultiUnits);
                 if (p == pchT)
                 {
                     this->RestoreMultiUnits(savedMultiUnits);
                     Assert(this->PeekFirst(p, last) != '.');
                     Error(ERRbadNumber);
                 }
                 Assert(!Js::NumberUtilities::IsNan(dbl));
-
+                if (likelyType == LikelyNumberType::BigInt)
+                {
+                    Assert(m_scriptContext->GetConfig()->IsESBigIntEnabled());
+                    token = tkBigIntCon;
+                    m_ptoken->SetBigInt(this->GetHashTbl()->PidHashNameLen(p, pchT, (uint32) (pchT - p)));
```

Comment:
Assert or fail fast here that pctT - p < UINT_MAX

Reply:
will do